pkg/crawler/log.go

```diff
@@ -17,7 +20,21 @@ func (c *Crawler) safePrintln(statusCode int, message string) {
 
 	if c.statusFilter.IsValid(c.FilterStatusQuery, int64(statusCode)) {
 		c.printMutex.Lock()
+		// Console log
 		_, _ = fmt.Fprintln(c.OutWriter, message)
+
+		// Write Json File if OutputJson flag is set
+		if c.OutputJson != "" {
+			c.writeJsonFile(message, c.OutputJson)
+		}
+
+		var url string
+		var data map[string]interface{}
+		if err := json.Unmarshal([]byte(message), &data); err != nil {
+			log.Fatal(err)
+		}
+		url = data["url"].(string)
+		c.writeToFileLine(url, c.OutputFile)
```

Comment:
When we write into a json file we should not also write to the text output file

Reply:
This seems like the solution. First of all, json will be checked. Otherwise it will write to outputFile.

My solution suggestion:
```go
// Write Json File if OutputJson flag is set
if c.OutputJson != "" {
	c.writeJsonFile(message, c.OutputJson)
} else {
	// Write to file if OutputFile flag is set
	if c.OutputFile != "" {
		var status string
		var url string
		var time string
		var duration string

		var data map[string]interface{}
		if err := json.Unmarshal([]byte(message), &data); err != nil {
			log.Fatal(err)
		}
		status = strconv.Itoa(data["status"].(int))
		url = data["url"].(string)
		time = strconv.Itoa(data["time"].(int))
		duration = strconv.Itoa(data["duration"].(int))

		c.writeLineToFIle(fmt.Sprintf("%s %s %s %s", status, url, time, duration), c.OutputFile)
	}
```

Reply:
In Go we usually prefer early returns over nesting so you could write:

```go
if c.OutputJson != "" {
    c.writeJsonFile(...)
    return
}

if c.OutputFile != "" {
    c.writeLineToFile(...)
    return
}

_, _ = fmt.Fprintln(c.OutWriter, message)
```
 I'd also pass the message to the function instead of doing the extraction here :)

Reply:
Yes, it looks better this way. At first, when I was writing it, I thought it would be good to have it available in different places.

Resolved.